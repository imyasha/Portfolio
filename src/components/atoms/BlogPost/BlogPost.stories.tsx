import { BlogPost, BlogPostProps } from './BlogPost';

export default {
	title: 'atoms/BlogPost',
	component: BlogPost,
};

export const BlogPostStory = (args: BlogPostProps) => <BlogPost {...args} />;

BlogPostStory.storyName = 'BlogPost';
BlogPostStory.args = {
	post: {
		title: 'Blog Post',
		description: 'Blog Description',
		slug: 'blog-post',
		tags: ['blog', 'post'],
		date: '2021-01-01',
	},
};
