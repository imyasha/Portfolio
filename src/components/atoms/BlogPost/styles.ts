import styled from 'styled-components';

const StyledBlogPost = styled.article`
	margin: 3rem 0;

	a {
		color: ${({ theme }) => theme.headline};
		text-decoration: none;
	}

	h1 {
		font-size: 1.5rem;
		margin: 0;
	}

	p {
		margin: 0;
	}
`;

const Flex = styled.div`
	display: flex;
	justify-content: space-between;
	flex-direction: column;

	@media screen and (min-width: 768px) {
		flex-direction: row;
	}
`;

const Tags = styled.ul`
	display: flex;
	flex-wrap: wrap;
	color: ${({ theme }) => theme.grey};
	margin-top: 0.5rem;

	li {
		margin-right: 1rem;
		font-size: 0.9rem;
	}
`;

export { Flex, StyledBlogPost, Tags };
