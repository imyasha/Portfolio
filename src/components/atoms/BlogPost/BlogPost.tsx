import { FC } from 'react';
import { format } from 'date-fns';
import { IPost } from '@Types';
import Link from 'next/link';
import { Flex, StyledBlogPost, Tags } from './styles';

export interface BlogPostProps {
	post: IPost;
}

const BlogPost: FC<BlogPostProps> = ({
	post: { title, description, slug, date, tags },
}) => {
	return (
		<StyledBlogPost>
			<Link href={`/blog/${slug}`}>
				<a>
					<Flex>
						<h1>{title}</h1>
						<div>{format(new Date(date), 'dd/MM/yyyy')}</div>
					</Flex>
					<p>{description}</p>
					<Tags>
						{tags.map((tag) => (
							<li key={tag}>#{tag}</li>
						))}
					</Tags>
				</a>
			</Link>
		</StyledBlogPost>
	);
};

export { BlogPost };
