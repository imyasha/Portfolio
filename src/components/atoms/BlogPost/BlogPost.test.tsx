import { BlogPost } from '../BlogPost';
import { render } from '@testing-library/react';

describe('BlogPost', () => {
	const props = {
		post: {
			title: 'Blog Post',
			description: 'Blog Description',
			slug: 'blog-post',
			tags: ['blog', 'post'],
			date: '2021-01-01',
		},
	};

	it('renders correctly', () => {
		const { container } = render(<BlogPost {...props} />);
		expect(container).toMatchSnapshot();
	});
});
