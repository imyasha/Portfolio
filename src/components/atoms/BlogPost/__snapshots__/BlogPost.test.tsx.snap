// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost renders correctly 1`] = `
<div>
  <article
    class="styles__StyledBlogPost-sc-1r22brh-0 fxlwgG"
  >
    <a
      href="/blog/blog-post"
    >
      <div
        class="styles__Flex-sc-1r22brh-1 ikIXMP"
      >
        <h1>
          Blog Post
        </h1>
        <div>
          01/01/2021
        </div>
      </div>
      <p>
        Blog Description
      </p>
      <ul
        class="styles__Tags-sc-1r22brh-2 iNbxpZ"
      >
        <li>
          #
          blog
        </li>
        <li>
          #
          post
        </li>
      </ul>
    </a>
  </article>
</div>
`;
